### PR TITLE
feat: Add WebView detection to AxeView

### PR DESCRIPTION
**Pull request by cyono**
Closes: #37 

See https://github.com/dequelabs/axe-android-app/pull/53 for an implementation.

## Requester Review Items

- [x] Ensure the Pull Request is into develop.
- [x] Test coverage has not gone down.
- [ ] Documentation updated or not needed. (Wiki, Issues, etc)
- [x] Angular Type leads to proper Semantic Version.

## Dev Team Review Items

To be filled out by @dequelabs/mobile.

- [ ] Ensure the Requester did not lie. Chastise them politely if they did.
- [ ] Code Quality review.
- [ ] Changes to Axe*.java classes are minimal, appropriate, and versioned properly.
  - [ ] Serialized Axe objects have not changed.
  - [ ] Public API has not changed.

## Code Owner Review Items

To be filled out by @dequelabs/mobileadmin.

- [ ] Any Rules package changes lead to proper Semantic Version.
- [ ] Security Review.

## Merger Review Items

- [ ] Ensure commit message matches title before squashing.

### CODE REVIEW

File: src/main/java/com/deque/axe/android/AxeView.java [hunk 1]
```diff
@@ -93,11 +94,26 @@
    */
   public final String viewIdResourceName;
 
+  /**
+   * Is view an instance of WebView.
+   */
+  private final boolean isWebView;
```
Comment (chriscm2006) [inline src/main/java/com/deque/axe/android/AxeView.java]: I feel like this property should be a calculated property, as it is not part of Android.View. We do make exceptions, but I'm not sure this should be one?

File: src/main/java/com/deque/axe/android/AxeView.java [hunk 2]
```diff
@@ -93,11 +94,26 @@
...
+
+  /**
+   * Is view a child of a WebView.
+   */
+  private final boolean isWebViewChild;
```
Comment (chriscm2006) [inline src/main/java/com/deque/axe/android/AxeView.java]: I feel like this should be renamed isInWebView. Also, I think we should just make it a function, I don't think the performance benefit of avoiding a  O(N) operation is worth serializing this.

We should at least make it transient.

File: src/main/java/com/deque/axe/android/AxeView.java [hunk 3]
```diff
@@ -159,6 +181,10 @@ private AxeView(
     this.text = text;
     this.viewIdResourceName = viewIdResourceName;
     this.children = children;
+    this.isWebView = isWebView;
+    this.isWebViewChild = isWebViewChild;
+
+    getDequeProps();
```
Comment (chriscm2006) [inline src/main/java/com/deque/axe/android/AxeView.java]: I don't understand this line.

File: src/main/java/com/deque/axe/android/AxeView.java (hunk 2 above)
Comment (cyono) [reply src/main/java/com/deque/axe/android/AxeView.java]: Or we can just remove this altogether and assume anything that is a child of a webview is in the webview.

File: src/main/java/com/deque/axe/android/AxeView.java (hunk 3 above)
Comment (cyono) [reply src/main/java/com/deque/axe/android/AxeView.java]: This relates to your first comment. There is a list of deque props that are the calculated values. This method populates the list based on the private fields `isWebView` and `isWebViewChild`.

I agree that we should clarify that we have calculated values as dequeProps, but I prefer not prepending deque to every field in the class that fits that criteria.

Comment (cyono) [reply src/main/java/com/deque/axe/android/AxeView.java]: Change to initDequeProps

**Comment by cyono** [comment]
Further refactoring of computed properties required. A larger PR will contain this same work and an updated Readme.

**Pull request closed without merging**